components/create_post/create_post.jsx

```diff
@@ -668,6 +668,9 @@ class CreatePost extends React.PureComponent {
     }
 
     focusTextbox = (keepFocus = false) => {
+        if (this.refs.textbox) {
```

Comment:
i don't understand the reasoning behind this. Can we not focus in the first place for read only channels in all brwosers?

Reply:
We can't for Firefox, it simply doesn't behave correctly when we do. I'm not 100% clear on the reasons, because this isn't documented anywhere and we have competing `focus()`'s throughout the webapp. Over about a week's time of experimenting on/off I narrowed it down to this - and I can't explain exactly why Firefox is acting weird, but suffice it to say Chrome/Safari don't have this problem. I'm wondering if this issue just goes away in future Firefox releases.

Reply:
I'm essentially unsetting the `focus()` here before re-focusing because I believe FF gets confused when the component changes to read-only while focus is still applied. Like I said, it's been really difficult to track down so this is a bit of black magic.

Reply:
I also renamed the ticket to match this new fix.